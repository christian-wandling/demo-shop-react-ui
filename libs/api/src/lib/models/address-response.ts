/* tslint:disable */
/* eslint-disable */
/**
 * Demo Shop API
 * A comprehensive API for managing an online store, providing endpoints for product catalog, user management, shopping cart operations, and order processing
 *
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { mapValues } from '../runtime';
/**
 *
 * @export
 * @interface AddressResponse
 */
export interface AddressResponse {
  /**
   *
   * @type {string}
   * @memberof AddressResponse
   */
  street: string;
  /**
   *
   * @type {string}
   * @memberof AddressResponse
   */
  apartment: string;
  /**
   *
   * @type {string}
   * @memberof AddressResponse
   */
  city: string;
  /**
   *
   * @type {string}
   * @memberof AddressResponse
   */
  region?: string | null;
  /**
   *
   * @type {string}
   * @memberof AddressResponse
   */
  zip: string;
  /**
   *
   * @type {string}
   * @memberof AddressResponse
   */
  country: string;
}

/**
 * Check if a given object implements the AddressResponse interface.
 */
export function instanceOfAddressResponse(value: object): value is AddressResponse {
  if (!('street' in value) || value['street'] === undefined) return false;
  if (!('apartment' in value) || value['apartment'] === undefined) return false;
  if (!('city' in value) || value['city'] === undefined) return false;
  if (!('zip' in value) || value['zip'] === undefined) return false;
  if (!('country' in value) || value['country'] === undefined) return false;
  return true;
}

export function AddressResponseFromJSON(json: any): AddressResponse {
  return AddressResponseFromJSONTyped(json, false);
}

export function AddressResponseFromJSONTyped(json: any, ignoreDiscriminator: boolean): AddressResponse {
  if (json == null) {
    return json;
  }
  return {
    street: json['street'],
    apartment: json['apartment'],
    city: json['city'],
    region: json['region'] == null ? undefined : json['region'],
    zip: json['zip'],
    country: json['country'],
  };
}

export function AddressResponseToJSON(json: any): AddressResponse {
  return AddressResponseToJSONTyped(json, false);
}

export function AddressResponseToJSONTyped(value?: AddressResponse | null, ignoreDiscriminator: boolean = false): any {
  if (value == null) {
    return value;
  }

  return {
    street: value['street'],
    apartment: value['apartment'],
    city: value['city'],
    region: value['region'],
    zip: value['zip'],
    country: value['country'],
  };
}
