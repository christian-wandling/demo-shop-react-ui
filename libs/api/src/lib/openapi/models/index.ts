/* tslint:disable */
 
export * from './add-cart-item-request';
export * from './address-response';
export * from './cart-item-response';
export * from './image-response';
export * from './order-item-response';
export * from './order-list-response';
export * from './order-response';
export * from './order-status';
export * from './product-list-response';
export * from './product-response';
export * from './shopping-session-response';
export * from './update-cart-item-quantity-request';
export * from './update-user-address-request';
export * from './update-user-phone-request';
export * from './user-response';
