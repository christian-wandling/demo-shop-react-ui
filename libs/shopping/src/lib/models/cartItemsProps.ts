import { CartItemResponse } from '@demo-shop-react-ui/api';

export interface CartItemsProps {
  /**
   * The list of items in the cart to be displayed.
   */
  items: CartItemResponse[];

  /**
   * Callback fired when a user requests to remove an item from the cart.
   * The callback receives the ID of the item to be removed.
   */
  onRemoveItem: (itemId: number) => void;

  onThumbnailClick?: () => void;
}
