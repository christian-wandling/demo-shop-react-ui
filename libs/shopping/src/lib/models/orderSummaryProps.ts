import { CartItemResponse } from '@demo-shop-react-ui/api';

export interface OrderSummaryProps {
  items: Array<CartItemResponse>;
  totalPrice: number;
  checkoutButtonEnabled: boolean;
  onRemoveItem: (id: number) => void;
  onCheckout: () => void;
}
