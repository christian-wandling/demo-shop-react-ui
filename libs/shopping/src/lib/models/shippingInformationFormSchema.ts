import { z } from 'zod';

export const shippingInformationFormSchema = z.object({
  firstname: z.string().min(1, 'First name is required').readonly(),
  lastname: z.string().min(1, 'Last name is required').readonly(),
  email: z.string().email('Invalid email address').readonly(),
  phone: z.string().nullable(),
  address: z
    .object({
      street: z.string().min(1, 'Street is required'),
      apartment: z.string().min(1, 'Apartment is required'),
      city: z.string().min(1, 'City is required'),
      region: z.string(),
      zip: z.string().min(1, 'Zip is required'),
      country: z.string().min(1, 'Country is required'),
    })
    .required(),
});

export type ShippingInformationFormData = z.infer<typeof shippingInformationFormSchema>;
