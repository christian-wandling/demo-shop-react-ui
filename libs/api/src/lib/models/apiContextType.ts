import { OrderApi, ProductApi, ShoppingSessionApi, UserApi } from '../openapi';

export type ApiContextType = {
  orderApi: OrderApi;
  productApi: ProductApi;
  shoppingSessionApi: ShoppingSessionApi;
  userApi: UserApi;
};
