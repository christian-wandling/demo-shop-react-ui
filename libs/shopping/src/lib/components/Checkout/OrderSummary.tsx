import { formatCurrency } from '@demo-shop-react-ui/shared';
import { CartItems } from '../CartItems/CartItems';
import { OrderSummaryProps } from '../../models/orderSummaryProps';

export const OrderSummary = ({
  items,
  totalPrice,
  onRemoveItem,
  onCheckout,
  checkoutButtonEnabled,
}: OrderSummaryProps) => {
  return (
    <section>
      <h2 className="text-lg font-medium text-gray-900" id="slide-over-title">
        Order Summary
      </h2>

      <CartItems items={items} onRemoveItem={onRemoveItem} />

      <div className="border-t border-gray-900/10 mt-12">
        <div className="mt-8 flex justify-between">
          <span>Total</span>
          <span>{formatCurrency(totalPrice)}</span>
        </div>
      </div>

      <section>
        <div className="mt-6 flex items-center justify-center gap-x-6">
          <button
            onClick={onCheckout}
            disabled={!checkoutButtonEnabled}
            className={`btn-checkout mt-6 lg:max-w-60 flex w-full items-center justify-center rounded-md border border-transparent px-8 py-3 text-base font-medium ${
              !checkoutButtonEnabled
                ? 'bg-gray-200 text-gray-800 pointer-events-none'
                : 'bg-indigo-600 text-white hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2'
            }`}>
            Checkout
          </button>
        </div>
      </section>
    </section>
  );
};
