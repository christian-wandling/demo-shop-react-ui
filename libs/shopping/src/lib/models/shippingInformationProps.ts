import { UseFormReturn } from 'react-hook-form';
import { ShippingInformationFormData } from './shippingInformationFormSchema';

export interface ShippingInformationProps {
  methods: UseFormReturn<ShippingInformationFormData>;
  onUpdateUser: () => void;
}
