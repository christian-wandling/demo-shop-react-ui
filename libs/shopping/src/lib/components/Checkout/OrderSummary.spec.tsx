import { fireEvent, render, screen } from '@testing-library/react';
import { OrderSummary } from './OrderSummary';
import { formatCurrency } from '@demo-shop-react-ui/shared';
import { OrderSummaryProps } from '../../models/orderSummaryProps';

// Mock dependencies
vi.mock('@demo-shop-react-ui/shared', () => ({
  formatCurrency: vi.fn(value => `$${value.toFixed(2)}`),
}));

vi.mock('../CartItems/CartItems', () => ({
  CartItems: vi.fn(() => <div data-testid="cart-items-mock" />),
}));

describe('OrderSummary', () => {
  const defaultProps: OrderSummaryProps = {
    items: [
      {
        id: 1,
        quantity: 1,
        productId: 1,
        productName: 'productName',
        productThumbnail: 'productThumbnail',
        unitPrice: 10,
        totalPrice: 10,
      },
    ],
    totalPrice: 10.99,
    onRemoveItem: vi.fn(),
    onCheckout: vi.fn(),
    checkoutButtonEnabled: true,
  };

  beforeEach(() => {
    vi.clearAllMocks();
  });

  it('renders the component with correct title', () => {
    render(<OrderSummary {...defaultProps} />);
    expect(screen.getByText('Order Summary')).toBeTruthy();
  });

  it('matches the snapshot', () => {
    const { baseElement } = render(<OrderSummary {...defaultProps} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('displays the formatted total price', () => {
    render(<OrderSummary {...defaultProps} />);

    expect(screen.getByText('Total')).toBeTruthy();
    expect(formatCurrency).toHaveBeenCalledWith(defaultProps.totalPrice);
    expect(screen.getByText('$10.99')).toBeTruthy();
  });

  it('renders enabled checkout button when checkoutButtonEnabled is true', () => {
    render(<OrderSummary {...defaultProps} />);

    const checkoutButton: HTMLButtonElement = screen.getByText('Checkout');
    expect(checkoutButton).toBeTruthy();
    expect(checkoutButton.disabled).toBeFalsy();
    expect(checkoutButton.classList).toContain('bg-indigo-600');
    expect(checkoutButton.classList).not.toContain('pointer-events-none');
  });

  it('renders disabled checkout button when checkoutButtonEnabled is false', () => {
    render(<OrderSummary {...defaultProps} checkoutButtonEnabled={false} />);

    const checkoutButton: HTMLButtonElement = screen.getByText('Checkout');
    expect(checkoutButton).toBeTruthy();
    expect(checkoutButton.disabled).toBeTruthy();
    expect(checkoutButton.classList).toContain('bg-gray-200');
    expect(checkoutButton.classList).toContain('pointer-events-none');
  });

  it('calls onCheckout when checkout button is clicked', () => {
    render(<OrderSummary {...defaultProps} />);

    const checkoutButton = screen.getByText('Checkout');
    fireEvent.click(checkoutButton);

    expect(defaultProps.onCheckout).toHaveBeenCalledTimes(1);
  });

  it('does not call onCheckout when disabled button is clicked', () => {
    render(<OrderSummary {...defaultProps} checkoutButtonEnabled={false} />);

    const checkoutButton = screen.getByText('Checkout');
    fireEvent.click(checkoutButton);

    expect(defaultProps.onCheckout).not.toHaveBeenCalled();
  });

  it('renders with zero total price', () => {
    render(<OrderSummary {...defaultProps} totalPrice={0} />);

    expect(formatCurrency).toHaveBeenCalledWith(0);
    expect(screen.getByText('$0.00')).toBeTruthy();
  });
});
