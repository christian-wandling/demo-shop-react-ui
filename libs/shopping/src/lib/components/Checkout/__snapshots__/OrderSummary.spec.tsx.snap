// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OrderSummary > matches the snapshot 1`] = `
<body>
  <div>
    <section>
      <h2
        class="text-lg font-medium text-gray-900"
        id="slide-over-title"
      >
        Order Summary
      </h2>
      <div
        data-testid="cart-items-mock"
      />
      <div
        class="border-t border-gray-900/10 mt-12"
      >
        <div
          class="mt-8 flex justify-between"
        >
          <span>
            Total
          </span>
          <span>
            $10.99
          </span>
        </div>
      </div>
      <section>
        <div
          class="mt-6 flex items-center justify-center gap-x-6"
        >
          <button
            class="btn-checkout mt-6 lg:max-w-60 flex w-full items-center justify-center rounded-md border border-transparent px-8 py-3 text-base font-medium bg-indigo-600 text-white hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2"
          >
            Checkout
          </button>
        </div>
      </section>
    </section>
  </div>
</body>
`;
